### DISCUSSION TITLE
Option to convert installed Type2 AppImages to Type3 (i.e. no "libfuse2" needed)?

### DISCUSSION THREAD

**Discussion by ivan-hc** [ivan-hc/AM#260] [Show and tell]
I don't know whether to take action myself or let the developers upstream take action (perhaps through your feedback in their repositories), but I already have an option in mind to convert the installed Type2 AppImages to Type3, i.e. which do not require `libfuse2` installed.

For my part, all of the 60+ x86_64 AppImages I created are now Type3.

I already know how to do it, I recently added a couple of commands in the "files.am" module that can locate them. It would be really easy for me to make a specific command... but I risk de-responsibilising the owners of their packages.

The thing that surprised me most of all, however, is that AppImages that can be updated via "zsync" (for example Libreoffice) if converted to Type3 can receive updates without losing their Type3 status.

I'm tempted to do it. But I'll start by saying that it doesn't work with all AppImages. I have only tested with the official AppImages of Libreoffice, Kdenlive (works) and Avidemux (doesn't work). There is also the fact that with each update, if it does not depend on "zsync", our Type3 will be replaced by the Type2 still used by the owner of the software. This is the only point that makes me doubt the usefulness of this new option that I could do.

Let me know.

EDIT: this is a test about the conversion of Kdenlive (I use custom actions in Thunar, on XFCE)

https://github.com/ivan-hc/AM/assets/88724353/506132a1-0e3f-415f-a299-5e178f8d81c4

@zen0bit @nazdridoy what do you think about? Shoul I implement a new "nolibfuse" option or is useless due to what I said above?

**Comment by ivan-hc** [comment]
Too late folks https://github.com/ivan-hc/AM/releases/tag/6.1

**Comment by ivan-hc** [comment]
Attention please! A critical bug in the new `nolibfuse.am` module has been discovered.

Update as soon as possible!

Please, see https://github.com/ivan-hc/AM/commit/569c66718d87c660337ff16b7c826f22ae27d669

**Reply by zen0bit** [reply]
ehm, ehm. Needs to be said again?

## Linting and test branches (releases)
Don't install/update from branch, update/install from stable or beta releases (for beta testers)

You are working on script which is actually used. Treat with care!!

**Reply by ivan-hc** [reply]
this is what I must do. Its the first time in three years I'm on Github that I have to face such a bug.
